test/framework/deployment_helpers.go

```diff
@@ -245,3 +250,139 @@ func WaitForDNSUpgrade(ctx context.Context, input WaitForDNSUpgradeInput, interv
 		return false, nil
 	}, intervals...).Should(BeTrue())
 }
+
+type DeployUnevictablePodInput struct {
+	WorkloadClusterProxy ClusterProxy
+	ControlPlane         *controlplanev1.KubeadmControlPlane
+	DeploymentName       string
+	Namespace            string
+
+	WaitForDeploymentAvailableInterval []interface{}
+}
+
+func DeployUnevictablePod(ctx context.Context, input DeployUnevictablePodInput) {
+	Expect(input.DeploymentName).ToNot(BeNil(), "Need a deployment name in DeployUnevictablePod")
+	Expect(input.Namespace).ToNot(BeNil(), "Need a namespace in DeployUnevictablePod")
+	Expect(input.WorkloadClusterProxy).ToNot(BeNil(), "Need a workloadClusterProxy in DeployUnevictablePod")
+	workloadClient := input.WorkloadClusterProxy.GetClientSet()
+
+	log.Logf("Check if namespace %s exists", input.Namespace)
+	if _, err := workloadClient.CoreV1().Namespaces().Get(ctx, input.Namespace, metav1.GetOptions{}); err != nil {
+		workloadClient.CoreV1().Namespaces().Create(ctx, &corev1.Namespace{
```

Comment:
Seems like from the CI failure, we need to check the err here.

Reply:
Yes, I have fixed it. Thanks